proxy.go

```diff
@@ -358,6 +362,29 @@ func (rp *reverseProxy) serveFromCache(s *scope, srw *statResponseWriter, req *h
 	}
 }
 
+func calcQueryParamsHash(origParams url.Values) uint32 {
+	var queryParamsHash uint32
+	var queryParams []config.Param
+	for param := range origParams {
+		if strings.HasPrefix(param, "param_") {
+			queryParams = append(queryParams, config.Param{
+				Key:   param,
+				Value: origParams.Get(param),
+			})
+		}
+	}
+	if queryParams == nil {
+		return 0
+	}
+	queryParamsRegistry, err := newParamsRegistry(queryParams)
```

Comment:
Can you compute hash without coupling to `newParamsRegistry`? 
You could extract useful part of that function to `utils`:

```
...
        h := fnv.New32a()
	for _, p := range params {
		str := fmt.Sprintf("%s=%s&", p.Key, p.Value)
		h.Write([]byte(str))
	}
```
and reuse it in both places

Reply:
@Garnek20 
I exported this logic to another function, called `calcMapHash`.

Hope you like it.

Reply:
Thanks! It looks good :)

Reply:
@Garnek20 Assuming you approve and merge the PR. I could not find the deployment schedule for CHProxy and would like to know when it will be available for the public use.

Reply:
Sure @giladl-neosec, once merged we'll release a new version 🙂